lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -91,11 +91,13 @@ sub list_ajax {
         prefetch => [qw(children parents)],
     )->all;
 
+    my $job_template_names = $self->schema->resultset('JobSettings')->get_job_template_names();
     # need to use all as the order is too complex for a cursor
     my $comment_count = $self->prefetch_comment_counts([map { $_->id } @jobs]);
     my @list;
     for my $job (@jobs) {
-        my $job_id = $job->id;
+        my $job_id    = $job->id;
+        my $test_name = $job_template_names->{$job_id} || $job->TEST;
```

Comment:
I think this is basically also what @Martchus said: You are repeating multiple times this logic to read out the job template name if it is defined and fall back to the "test" value. At the very least this should be directly a method of `$job` itself.

As an alternative: How about setting "TEST" accordingly to the "job template name" if it is defined on job creation and not reading out job template name everywhere?

Reply:
I think @Amrysliu mentioned that alternative herself some time ago and wanted to look into it. (It might have an unintended impact considering that "TEST" is used in several places.)